agent.py

```diff
@@ -196,6 +206,266 @@ def _do_restart(self):
+                "A useful external tool for yaml parsing can be found at "
+                "http://yaml-online-parser.appspot.com/")
+        return 1
+
+class Flare(object):
+    """
+    Compress all important logs and configuration files for debug,
+    and then send them to Datadog (which transfers them to Support)
+    """
+
+    DATADOG_SUPPORT_URL = '/zendesk/flare'
+    PASSWORD_REGEX = re.compile('( *(\w|_)*pass(word)?:).+')
+    COMMENT_REGEX = re.compile('^ *#.*')
+    APIKEY_REGEX = re.compile('^api_key:')
+
+    def __init__(self, cmdline=False, case_id=None):
+        self._case_id = case_id
+        self._cmdline = cmdline
+        self._init_tarfile()
+        self._save_logs_path(get_logging_config())
+        config = get_config()
+        self._api_key = config.get('api_key')
+        self._url = "{0}{1}".format(config.get('dd_url'), self.DATADOG_SUPPORT_URL)
+        self._hostname = get_hostname(config)
+        self._prefix = "datadog-{0}".format(self._hostname)
+
+    def collect(self):
+        if not self._api_key:
+            raise Exception('No api_key found')
+        self._print("Collecting logs and configuration files:")
+
+        self._add_logs_tar()
+        self._add_conf_tar()
+        self._print("  * datadog-agent configcheck output")
+        self._add_command_output_tar('configcheck.log', configcheck)
+        self._print("  * datadog-agent status output")
+        self._add_command_output_tar('status.log', self._supervisor_status)
+        self._print("  * datadog-agent info output")
+        self._add_command_output_tar('info.log', self._info_all)
+
+        self._print("Saving all files to {0}".format(self._tar_path))
+        self._tar.close()
+
+    # Upload the tar file
+    def upload(self, confirmation=True):
+        # Ask for confirmation first
+        if confirmation:
+            self._ask_for_confirmation()
+
+        email = self._ask_for_email()
+
+        self._print("Uploading {0} to Datadog Support".format(self._tar_path))
+        url = self._url
+        if self._case_id:
+            url = "{0}/{1}".format(self._url, str(self._case_id))
+        files = {'flare_file': open(self._tar_path, 'rb')}
+        data = {
+            'api_key': self._api_key,
+            'case_id': self._case_id,
+            'hostname': self._hostname,
+            'email': email
+        }
+        r = requests.post(url, files=files, data=data)
+        self._analyse_result(r)
+
+    # Start by creating the tar file which will contain everything
+    def _init_tarfile(self):
+        # Default temp path
+        self._tar_path = os.path.join(tempfile.gettempdir(), 'datadog-agent.tar.bz2')
+
+        if os.path.exists(self._tar_path):
+            os.remove(self._tar_path)
+        self._tar = tarfile.open(self._tar_path, 'w:bz2')
+
+    # Save logs file paths
+    def _save_logs_path(self, config):
+        prefix = ''
+        if get_os() == 'windows':
```

Comment:
Also you should probably use the get_logging_config function https://github.com/DataDog/dd-agent/blob/master/config.py#L900